src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -252,6 +256,15 @@ protected Query doToQuery(QueryShardContext context) {
             );
         }
 
+        byte[] byteVector = new byte[0];
+        if (VectorDataType.BYTE.equals(vectorDataType)) {
+            byteVector = new byte[vector.length];
+            for (int i = 0; i < vector.length; i++) {
+                validateByteVectorValue(vector[i]);
+                byteVector[i] = (byte) vector[i];
```

Comment:
Why can we cast like this?

Reply:
To convert it from float to byte type. The value will still remain the same after casting because we have already validated to make sure it is within the byte range and without any decimal values.